### PR TITLE
steps: add 4.1.0 and cleanup

### PR DESCRIPTION
**Pull request by tristan0x**
* add new version 4.1.0
* remove unused versions and cleanup dependencies accordingly
* rename CMake option `STEPS_USE_VESICLE_MODEL` to `STEPS_USE_VESICLE_SOLVER`
* use spack function `define_from_variant` for `USE_MPI` and `USE_PETSC` since both master and 4.1.0 now allow boolean values

### CODE REVIEW

**Comment by tristan0x** [comment]
@matz-e 
https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/643787#L2317
I don't have `+=`  in the steps recipe :cry:

**Comment by matz-e** [comment]
> @matz-e https://bbpgitlab.epfl.ch/hpc/spack/-/jobs/643787#L2317 I don't have `+=` in the steps recipe cry

BSD ticket? :grimacing: 

/jk, looking into it.

File: bluebrain/repo-patches/packages/steps/package.py [hunk 1]
```diff
@@ -14,18 +14,16 @@ class Steps(CMakePackage):
...
+    maintainers("tristan0x")
 
-    version("develop", branch="master")
-    version("5.0.0a", commit="d75d46f")
-    version("4.0.0", tag="4.0.0", preferred=True)
-    version("3.6.0", tag="3.6.0")
-    version("3.5.0b", commit="b2be5fe")
+    version("develop", branch="master", submodules=True)
+    version("5.0.0a", commit="b4bdd64", submodules=True)
+    version("4.1.0", submodules=True, preferred=True)
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/steps/package.py]:
```suggestion
    submodules = True

    version("develop", branch="master")
    version("5.0.0a", commit="b4bdd64")
    version("4.1.0", tag="4.1.0", preferred=True)
```
Also why does this not have a tag?

**Review by matz-e — COMMENTED** [review]
I have a hunch that the `+=` may come from the fetcher for 4.1.0:

https://github.com/BlueBrain/spack/blob/develop/lib/spack/spack/fetch_strategy.py#L802-L809

You don't set a ref... so there's a `None` on the right. Just a hypothesis, but seems appropriate.

File: bluebrain/repo-patches/packages/steps/package.py (hunk 1 above)
Comment (tristan0x) [reply bluebrain/repo-patches/packages/steps/package.py]: > Also why does this not have a tag?

Why isn't it the default? 😆 
/jk, will report this upstream. There should be a nice error message for retards like me.

**Review by matz-e — APPROVED** [review]

File: bluebrain/repo-patches/packages/steps/package.py [hunk 2]
```diff
@@ -55,40 +56,37 @@ class Steps(CMakePackage):
...
     depends_on("pkgconfig", type="build")
-    depends_on("py-build", type="build", when="@5:")
-    depends_on("py-pip", type="build", when="@5:")
     depends_on("py-cython")
-    depends_on("py-h5py", type=("build", "test", "run"))
     depends_on("py-gcovr", when="+coverage", type="build")
+    depends_on("py-h5py", type=("build", "test", "run"))
+    depends_on("py-pip", type="build", when="@5:")
     depends_on("py-matplotlib", type=("build", "test"))
+    depends_on("py-build", type="build", when="@5:")
```
Comment (matz-e) [inline bluebrain/repo-patches/packages/steps/package.py:83]: Silly question, what's the order here? Not alphabetic...

**Pull request merged**